vizro-core/src/vizro/models/_page.py

```diff
@@ -132,11 +132,11 @@ def build(self):
         ]
         return self._make_page_layout(controls_content, components_content)
 
-    def _update_graph_theme(self):
+    def _update_graph_theme(self):  # LN: needs to be refactored so plotly-independent or extendable - DONE
         outputs = [
-            Output(component.id, "figure", allow_duplicate=True)
+            component._get_update_graph_theme_output()
             for component in self.components
-            if isinstance(component, Graph)
+            if hasattr(component, "_get_update_graph_theme_output")
         ]
         if outputs:
```

Comment:
@maxschulz-COL  - can this be moved completely out of the page model? Asking because this is currently affecting the creation of tabs/containers in a draft PR I have. Optimally, this should live in the graph itself, as these containers shouldn't need to be responsible for the theme update in their components.

Reply:
hm i can try

Reply:
Yes, this is actually quite easy I think, and definitely the right call!